core/src/main/scala/quasar/destination/avalanche/AvalancheDestination.scala

```diff
@@ -0,0 +1,274 @@
+
+      _ <- deleteService(BlobPath(List(PathElem(freshName.value))))
+    } yield ()
+
+  private def copy(tableName: TableName, freshName: FileName, cols: NonEmptyList[Fragment])
+      : F[Unit] = {
+
+    val dropQuery = dropTableQuery(tableName).update
+
+    val createQuery = createTableQuery(tableName, cols).update
+
+    val loadQuery = copyQuery(tableName, freshName).update
+
+    for {
+      _ <- debug(s"Table creation query:\n${createQuery.sql}")
+
+      _ <- debugRedacted(s"Load query:\n${loadQuery.sql}")
+
+      _ <- debug(s"Drop table query:\n${dropQuery.sql}")
+
+      load = Sync[ConnectionIO].bracket(createStatement)(st =>
+        for {
+          _ <- Sync[ConnectionIO].delay(st.execute(loadQuery.sql))
+          count <- Sync[ConnectionIO].delay(st.getUpdateCount)
+        } yield count)(st => Sync[ConnectionIO].delay(st.close))
+
+      count <- (dropQuery.run *> createQuery.run *> load).transact(xa)
+
+      _ <- debug(s"Finished load")
+
+      _ <- debug(s"Load result count: $count")
+
+      _ <- debug(s"Finished table copy")
+
+    } yield ()
+  }
+
+  private def upload(bytes: Stream[F, Byte]): F[FileName] =
+    for {
+      freshNameSuffix <- Sync[F].delay(UUID.randomUUID().toString)
+      freshName = s"reform-$freshNameSuffix"
+
+      _ <- refreshToken
+
+      containerURL <- refContainerURL.get
+
+      put = AzurePutService.mk[F](containerURL.value)
+
+      _ <- debug(s"Starting upload of $freshName")
+
+      _ <- put((BlobPath(List(PathElem(freshName))), bytes))
+
+      _ <- debug(s"Finished upload of $freshName")
+
+    } yield FileName(freshName)
+
+  // Ingres accepts double quotes as part of identifiers, but they must
+  // be repeated twice This implies identifiers with an even number of
+  // double quotes are valid and those with an odd number of double
+  // quotes are invalid.
+  // More details:
+  // https://docs.actian.com/avalanche/index.html#page/SQLLanguage%2FRegular_and_Delimited_Identifiers.htm%23ww414482
+  private def evenQuotes(tableName: String): Boolean =
+    if (tableName.toList.count(_ === '"') % 2 === 0)
+      true
+    else
+      false
+
+  private def removeSingleQuotes(str: String): String =
+    str.replace("'", "")
+
+  private def ensureValidTableName(r: ResourcePath): F[TableName] =
+    r match {
+      case file /: ResourcePath.Root if evenQuotes(file) => TableName(file).pure[F]
+      case _ => MonadResourceErr[F].raiseError(ResourceError.notAResource(r))
+    }
+
+  private def ensureValidColumns(columns: List[TableColumn]): Either[String, NonEmptyList[Fragment]] =
+    for {
+      cols0 <- columns.toNel.toRight("No columns specified")
+      cols <- cols0.traverse(mkColumn(_)).toEither.leftMap(errs =>
+        s"Some column types are not supported: ${mkErrorString(errs.asScalaz)}")
+    } yield cols.asScalaz
+
+  private def copyQuery(tableName: TableName, fileName: FileName): Fragment = {
+    val table = removeSingleQuotes(tableName.name)
```

Comment:
Looks like `removeSingleQuotes` is called here but not in eg `createTableQuery`. That means that if a table name like `I'm using a single quote` is specified then the table names wouldn't match and the job would fail.

Reply:
Good observation. I shouldn’t be calling this here. At all.